distribution/lib/Standard/Base/0.0.0-dev/src/Widget_Helpers.enso

```diff
@@ -34,8 +45,10 @@ make_date_format_selector (date:Date=(Date.new 2012 3 14)) =
     formats = ['d/M/yyyy', 'dd/MM/yyyy', 'd-MMM-yy', 'd MMMM yyyy', 'M/d/yyyy', 'MM/dd/yyyy', 'MMMM d, yyyy', 'yyyy-MM'].map f->
         [f + " (e.g. " + date.format f + ")", f.pretty]
     custom_locale_format =
-        format = Date_Time_Formatter.from 'd MMMM yyyy' locale=Locale.france
-        ['d MMMM yyyy - with custom Locale (e.g. ' + date.format format + ')', "(Date_Time_Formatter.from 'd MMMM yyyy' locale=Locale.france)"]
+        format = Date_Time_Formatter.from_simple_pattern 'd MMMM yyyy' locale=Locale.france
+        label = 'd MMMM yyyy - with custom Locale (e.g. ' + date.format format + ')'
+        code = "(Date_Time_Formatter.from_simple_pattern 'd MMMM yyyy' locale=Locale.france)"
+        Option label code [["pattern", make_single_choice formats], ["locale", Locale.default_widget]]
```

Comment:
Shouldn't these nested widgets be set automatically?

Or if we set the 'toplevel' widget, than all inner ones must also be set manually to work?

Asking mostly to understand and use the correct practice myself when adding any widgets.

Reply:
Yes the top level ones will be set automatically (when a bug is fixed).
However in this case I am deliberately adding the inner ones so I can be consistent with outer one and be value type determining what is added.